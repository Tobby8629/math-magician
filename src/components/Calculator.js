import React, { useState } from 'react';
import calculate from './logic/calculate';

const Calculator = () => {
  const digits = ['AC', '+/-', '%', '\u00f7', '7', '8', '9', 'x', '4', '5', '6', '-', '1', '2', '3', '+', '0', '.', '='];
  const [solution, setsolution] = useState({ total: null, next: null, operation: null });
  const calc = (e) => {
    setsolution(calculate(solution, e.target.textContent));
  };
  return (
    <div className="calculator">
      <div className="digits">

        <div className="input" data-testid="screen">
          {solution.total}
          {solution.operation}
          {solution.next}
        </div>

        <div className="grid">
          {digits.map((digit) => (
            <div
              className="digit"
              key={digit}
              onClick={calc}
              role="button"
              tabIndex={0}
              onKeyDown={calc}
              data-testid="btn"
            >
              {digit}
            </div>
          ))}
        </div>

      </div>
    </div>
  );
};

export default Calculator;
