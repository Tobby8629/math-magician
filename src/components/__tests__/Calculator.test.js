import { fireEvent, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Calculator from '../Calculator';

describe('test the calculator component', () => {
  test('Calculator component match snapshot', () => {
    const tree = renderer.create(<Calculator />).toJSON;
    expect(tree).toMatchSnapshot();
  });

  test('calculator display fired button', () => {
    render(<Calculator />);
    const show = screen.getByTestId('screen');
    const btn = screen.getAllByTestId('btn');
    fireEvent.click(btn[4]);
    expect(show).toHaveTextContent('7');
  });
});
