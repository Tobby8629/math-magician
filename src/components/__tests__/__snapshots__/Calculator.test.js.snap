// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test the calculator component Calculator component match snapshot 1`] = `[Function]`;
